core/src/main/scala/spire/std/double.scala

```diff
@@ -17,147 +17,14 @@ package spire
-}
-
-trait DoubleIsTrig extends Trig[Double] {
-  def e: Double = Math.E
-  def pi: Double = Math.PI
-
-  def exp(a: Double): Double = Math.exp(a)
-  def expm1(a: Double): Double = Math.expm1(a)
-  def log(a: Double): Double = Math.log(a)
-  def log1p(a: Double): Double = Math.log1p(a)
-
-  def sin(a: Double): Double = Math.sin(a)
-  def cos(a: Double): Double = Math.cos(a)
-  def tan(a: Double): Double = Math.tan(a)
-
-  def asin(a: Double): Double = Math.asin(a)
-  def acos(a: Double): Double = Math.acos(a)
-  def atan(a: Double): Double = Math.atan(a)
-  def atan2(y: Double, x: Double): Double = Math.atan2(y, x)
-
-  def sinh(x: Double): Double = Math.sinh(x)
-  def cosh(x: Double): Double = Math.cosh(x)
-  def tanh(x: Double): Double = Math.tanh(x)
-
-  def toRadians(a: Double): Double = (a * 2 * pi) / 360
-  def toDegrees(a: Double): Double = (a * 360) / (2 * pi)
-}
-
-trait DoubleOrder extends Order[Double] {
-  override def eqv(x: Double, y: Double): Boolean = x == y
-  override def neqv(x: Double, y: Double): Boolean = x != y
-  override def gt(x: Double, y: Double): Boolean = x > y
-  override def gteqv(x: Double, y: Double): Boolean = x >= y
-  override def lt(x: Double, y: Double): Boolean = x < y
-  override def lteqv(x: Double, y: Double): Boolean = x <= y
-  override def min(x: Double, y: Double): Double = Math.min(x, y)
-  override def max(x: Double, y: Double): Double = Math.max(x, y)
-  def compare(x: Double, y: Double): Int = java.lang.Double.compare(x, y)
-}
-
-trait DoubleSigned extends Signed[Double] with DoubleOrder {
-  def order = this
-  override def signum(a: Double): Int = Math.signum(a).toInt
-  override def abs(a: Double): Double = if (a < 0.0) -a else a
-}
-
-trait DoubleTruncatedDivision extends TruncatedDivisionCRing[Double] with DoubleSigned {
-  def toBigIntOpt(a: Double): Opt[BigInt] = if (a.isWhole) Opt(BigDecimal(a).toBigInt) else Opt.empty[BigInt]
-  def tquot(a: Double, b: Double): Double = (a - (a % b)) / b
-  def tmod(a: Double, b: Double): Double = a % b
-}
-
-trait DoubleIsReal extends IsRational[Double] with DoubleTruncatedDivision {
-  def toDouble(x: Double): Double = x
-  def ceil(a: Double): Double = Math.ceil(a)
-  def floor(a: Double): Double = Math.floor(a)
-  def round(a: Double): Double = spire.math.round(a)
-  def isWhole(a: Double): Boolean = a % 1.0 == 0.0
-  def toRational(a: Double): Rational = Rational(a)
-}
 
 @SerialVersionUID(0L)
 class DoubleAlgebra extends DoubleIsField with DoubleIsNRoot with DoubleIsTrig with DoubleIsReal with Serializable
 
 trait DoubleInstances {
-  implicit final val DoubleAlgebra: Field[Double]
-    with NRoot[Double]
-    with Trig[Double]
-    with IsRational[Double]
-    with TruncatedDivisionCRing[Double]
-    with Signed[Double]
-    with Order[Double] = new DoubleAlgebra
+  implicit final val DoubleAlgebra
+    : DoubleIsField with DoubleIsNRoot with DoubleIsTrig with DoubleIsReal with DoubleSigned with Serializable =
+    new DoubleAlgebra
```

Comment:
The breaking change was this. You can't replace `Field[Double]` with `DoubleIsField`, this breaks bincompat.

Reply:
To make the change bin-compatibly:

```scala
// keep the old one, but remove `implicit`
final val DoubleAlgebra: Field[Double]
    with NRoot[Double]
    with Trig[Double]
    with IsRational[Double]
    with TruncatedDivisionCRing[Double]
    with Signed[Double]
    with Order[Double] = new DoubleAlgebra

// can't be val b/c bincompat
// hacky but it works
implicit final def DoubleAlgebraBincompat: DoubleAlgebra = DoubleAlgebra.asInstanceOf[DoubleAlgebra]
```

Reply:
Unfortunately, if I don't do that, scala-3 won't pick up the `inline` defs and the desired inlining does not happen.

Reply:
So the price of getting this improvement is breaking ABI

Reply:
It's OK with me if that's something you'd prefer to not do :)

Reply:
Sorry I'm not sure I understand. You tried my change and it didn't work?

Reply:
Oh I missed your proposed fix, I will try that